content/automate/ManageIQ/System/CommonMethods/Utils.class/__methods__/log_object.rb

```diff
@@ -18,6 +19,28 @@ def self.log_and_exit(msg, exit_code, handle = $evm)
               handle.log('info', "Script ending #{msg} code : #{exit_code}")
               exit(exit_code)
             end
+
+            # If you want to create a notification and log a message without specifying a handle using the global $evm
+            #    ManageIQ::Automate::System::CommonMethods::Utils::LogObject.log_and_notify(:level, msg, subject)
+            #
+            # If you want to create a notification and log a message using a specific handle
+            #    ManageIQ::Automate::System::CommonMethods::Utils::LogObject.log_and_notify(:level, msg, subject, handle)
+            #
+            def self.log_and_notify(level, message, subject, handle = $evm)
+              handle.create_notification(:level => notify_level(level.downcase.to_s), :message => message, :subject => subject)
+              handle.log(log_level(level.downcase.to_s), message)
+            end
+
+            def self.log_level(level)
+              log_level = level == "warning" ? "warn" : level
+              %w(info warn error).include?(log_level) ? log_level : "info"
+            end
+
+            def self.notify_level(level)
```

Comment:
@billfitzgerald0120 Instead of using a function can we just use a hash e.g
```ruby
NOTIFY_LEVEL_TO_LOG_LEVEL = { 
  'info'    => 'info',
  'warning' => 'warn',
  'error'   => 'error',
  'success' => 'info'
}
NOTIFY_LEVEL_TO_LOG_LEVEL.default = 'info'
```
To fetch values you could use something like
```ruby
puts NOTIFY_LEVEL_TO_LOG_LEVEL['fred']
puts NOTIFY_LEVEL_TO_LOG_LEVEL['warning']
```